doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,822 @@
+||SAI_STP_PORT_STATE_FORWARDING|
+||SAI_STP_PORT_STATE_BLOCKING|
+|VLAN STP Instance|SAI_VLAN_ATTR_STP_INSTANCE|
+|Switch STP Attributes|SAI_SWITCH_ATTR_DEFAULT_STP_INST_ID|
+||SAI_SWITCH_ATTR_MAX_STP_INSTANCE|
+
+## New SAI Attributes
+MSTP design requires one new attribute `SAI_HOSTIF_TRAP_TYPE_MSTP` for control trap packets which will be defined in saihostif.h.
+
+[SAI/saihostif.h](https://github.com/opencomputeproject/SAI/blob/master/inc/saihostif.h)
+
+# Additional Features
+
+## BPDU Guard
+BPDU guard feature will also be supported by MSTP. The details are mentioned in PVST HLD.
+
+## Root Guard
+Root guard feature will also be supported by MSTP. The details are mentioned in PVST HLD.
+
+## Edge Port
+Edge ports immediately transition to the forwarding state upon activation and do not participate in STP topology calculations. In PVST, this functionality is achieved through the "Portfast" feature, while in MSTP standard, the term "Edge Port" is used. To ensure user clarity, separate command will be provided for configuring Edge Ports.
+
+Portfast command will be disabled when the spanning-tree mode is set to `mstp` and Edge Port command will not function under `pvst` spanning-tree mode.
+
+Please note that the Edge Port functionality is equivalent to Portfast, with the only difference being the terminology and command usage based on the spanning-tree mode. The configuration for Edge ports will be stored in the portfast attribute in the database and the backend functionality remains the same as portfast.
+
+## Uplink Fast
+MSTP standard does not support uplink fast so uplink fast functionality will be disable for MSTP.
+
+
+#  Sequence Diagrams
+## MSTP global enable
+
+Only the VLANs that are currently present will be mapped to IST instance. A VLAN cannot be mapped to an instance if it has not been created yet.
+
+![MSTP Global Enable](images/MSTP_global_enable.png)
+
+## MSTP global disable
+![MSTP Global Disable](images/MSTP_global_disable.png)
+
+## MSTP region name/version change
+![MSTP Region Config](images/MSTP_region_config.png)
+
+## Instance creation
+![MSTP Instance Create](images/MSTP_instance_create.png)
+
+## Instance deletion
+![MSTP Instance Del](images/MSTP_instance_del.png)
+
+## Add VLAN to instance
+![MSTP VLAN Add](images/MSTP_vlan_add.png)
+
+## Del VLAN from instance
+![MSTP VLAN Del](images/MSTP_vlan_del.png)
+
+## Add VLAN member
+![MSTP VLAN Member Add](images/MSTP_vlan_member_add.png)
+
+## Del VLAN member
+![MSTP VLAN Member Del](images/MSTP_vlan_member_del.png)
+
+Update port-state, topology change and instance-interface events remain the same as depicted in [Sequence Diagrams of PVST](https://github.com/sandeep-kulambi/SONiC/blob/631ab18211e7e396b138ace561b7a04e7f7b49a1/doc/stp/SONiC_PVST_HLD.md#4-flow-diagrams).
+
+# Configuration Commands
+Following configuration commands will be provided for configuration of MSTP:
+## Global Level
+- **config spanning_tree {enable|disable} {mstp|pvst}**
```

Comment:
Are  we going to support pvst mode also? There are variations of PVST  like PVST+ and  RPVST+.  Do we support all the STP variations?

Reply:
@balajib-cisco , SONiC already has PVST mode, proposed in [HLD](https://github.com/sonic-net/SONiC/pull/386). We are proposing MSTP mode.

Reply:
AFAIK, PVST mode support is not committed/merged to SONiC community branch. Those PVST PRs are still open https://github.com/sonic-net/sonic-swss/pull/1058.  Are you going to commit those along with this MSTP commit? How are we going to coexist?

Reply:
Yes, there are some open PRs for PVST. BRCM will help with those PRs. We are adding MSTP mode along with existing STP functionalities. PVST & MSTP can coexist and used accordingly depending on usecases.